### DISCUSSION TITLE
FYSETC S6 with TMC2209

### DISCUSSION THREAD

**Discussion by hartmutgw2** [grblHAL/STM32F4xx#130] [Q&A]
Hello,

since weeks I try to get my MPCNC with Fysetc S6 V2.1 and Fysetc TMC2209 V3.1 drivers working.
Without TRINAMIC_ENABLE=2209 it works, however I cannot define a hold current for Z stepper and due to weight of the spindle this configuration doesn't work for me.

When TMC2209 is enabled the board does not show up with lsusb so no communication to controller.
Jumper for UART communication to stepper drivers is set on the board and the controller worked with klipper and tmc2209 support before so I assume it is not hardware related.

If I flash the non-Trinamic version it works again and I'm wondering what I've missed.

Anybody here with a similar configuration or any hint what I'm doing wrong?

Thanks,
Hartmut

**Comment by terjeio** [comment]
[Tentative fix](https://github.com/grblHAL/STM32F4xx/issues/131) committed as I do not have a board to verify with. Please report back in the linked issue if fix is working.

FYI the [board map](https://github.com/grblHAL/STM32F4xx/blob/master/Inc/fysetc_s6_map.h) was orginally provided by a user, my guess is he did not test with Trinamic drivers.
